src/FEValues/common_values.jl

```diff
@@ -30,28 +30,22 @@ end
 end
 
 """
-    reinit!(cv::CellValues, x::Vector)
-    reinit!(bv::FaceValues, x::Vector, face::Int)
+    reinit!(cv::CellValues, x::Vector, cell::Union{AbstractCell,Nothing}=nothing)
+    reinit!(bv::FaceValues, x::Vector, face::Int, cell::Union{AbstractCell,Nothing}=nothing)
 
 Update the `CellValues`/`FaceValues` object for a cell or face with coordinates `x`.
 The derivatives of the shape functions, and the new integration weights are computed.
+For interpolations with non-identity mappings, the current `cell` is also required. 
 """
 reinit!
```

Comment:
I think this is error prone to have cell=nothing per default (for users). Can we somehow control the reinit! by dispatches? Or at least some clear error message for the user when he does not pass the correct information?

Reply:
Yes, that is a good point - the error that is thrown now might be hard to understand otherwise.